### DISCUSSION TITLE
How to add credits info through CLI?

### DISCUSSION THREAD

**Discussion by Kaien715** [comictagger/comictagger#531] [Q&A] [answered]
I have spend  good two hours trying to figure out how to use CLI. Most stuff go pretty smoothly except for adding credit (penciller,writer...etc). Can anyone help me out?

Say I want to add penciller = ABC and writer = DEF in all files in the current directory:
ComicTagger -s -t cr -g -f -m "???????"

I know I need to start with credits=, but I have tried everything I can think of to fill role/person/primary in different kind of combination, and system just won't take it. If someone can provide me an example, it would be really greatful.

Thanks in advance!

**Comment by lordwelch** [comment, marked as answer]
You should be able to do `comictagger -s -t cr -g -f -m "credit=writer:DEF,credit=penciller:ABC"`

**Reply by Kaien715** [reply]
Thank you! You have saved me hours guessing. I thought I need ( ) or {} of some sort and need to supply all 3 arguements. And keep trying different combination. lol

**Reply by lordwelch** [reply]
If you want to set primary you would do `credit=writer:DEF:p` it doesn't matter what you put after the second `:` it just checks that something is there

**Comment by lordwelch** [comment]
For anyone wanting to do manual bulk edits via the CLI [comicbox] is much better at it than ComicTagger.

[comicbox]: https://github.com/ajslater/comicbox

**Reply by beville** [reply]
Regarding `comicbox` I was playing with it recently, and agree it's a great tool.  I ran into an issue, though.  When I modify a cbz file which already has a ComicInfo.xml with a line like:
```
comicbox  "Whiz Comics #007 (1940).cbz"  -m "Publisher: Fawcett Pub" -w cr
```
It replaces "ComicInfo.xml" with "comicinfo.xml".  If I then inspect the file with ComicTagger, it can't find the metadata.

I haven't been keeping up with the developements at the Anansi Project, so I'm not sure what the accepted standard is, and if ComicBox is writing the wrong filename, or if CT is not being flexible enough.

Do you see the same behavior?

**Reply by ajslater** [reply]
[Comicbox v1.1.1](https://github.com/ajslater/comicbox/releases/tag/v1.1.1) now writes ComicInfo.xml, MetronInfo.xml and CoMet.xml in TitleCase for the benefit of progams expecting that and human readability.

Comicbox continues to read metadata files in any case and from all levels of the archive directory tree.

**Reply by lordwelch** [reply]
I believe ComicTagger has always been case sensitive for ComicInfo.xml there isn't a real reason as far as I can tell. I just haven't gotten around to copying the CoMet.xml code for finding the filename over to the comicrack code

**Reply by beville** [reply]
My take is that since both zip and rar directories are both-case sensitive and case-retentive then "ComicInfo.xml" and "comicinfo.xml" are distinct files and can both exist in the same archive.   The ComicRack app chose to use the TitleCase version, and since there's no other standards discussed elsewhere, I'm inclined to think that should be the de facto standard for both readers and writers.

If a reader app wants be accommodating to accept ether or both files and do some clever merging, I guess that's fine.  But if not all apps do it, then you get inconsistent metadata being read in from different apps.  

Probably this could be fodder for discussion at the Anansi Project.

**Reply by lordwelch** [reply]
I'm inclined to agree CT has been around for a while and no one ever asked for it that I've seen.
